File: rb/features.rb
# YesAsAService SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module YesAsAServiceFeatures
  def self.make_feature(name)
    case name
    when "base"
      YesAsAServiceBaseFeature.new
    when "ratelimit"
      YesAsAServiceRatelimitFeature.new
    when "retry"
      YesAsAServiceRetryFeature.new
    when "test"
      YesAsAServiceTestFeature.new
    when "timeout"
      YesAsAServiceTimeoutFeature.new
    else
      YesAsAServiceBaseFeature.new
    end
  end
end

File: py/features.py
# YesAsAService SDK feature factory

from feature.base_feature import YesAsAServiceBaseFeature
from feature.test_feature import YesAsAServiceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YesAsAServiceBaseFeature(),
        "test": lambda: YesAsAServiceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
